### ISSUE TITLE
EWMH compliance: _NET_WM_STATE

### ISSUE THREAD

**Issue by theguy147** [leftwm/leftwm#152]
It appears that LeftWM only declares the _NET_WM_STATE property once a hint/atom is added but not before as *expected* (?).

In the [EWMH specs](https://specifications.freedesktop.org/wm-spec/wm-spec-1.3.html#idm45760170789728) it says
> **_NET_WM_STATE, ATOM[]**
A list of hints describing the window state. Atoms present in the list MUST be considered set, atoms not present in the list MUST be considered not set. [...]

It really does not state this in detail but I understand this as it implying "The list itself MUST also be set". For other tools to be able to check _NET_WM_STATE it seems to me that this list should be set from the beginning even if it is empty.

Here is an example of the issue with all steps to reproduce:

1.) start application e.g. `firefox` normally (not maximized or anything)
2.) check _NET_WM_STATE of application with `xprop` (and click on application window)
```bash
$ xprop _NET_WM_STATE
_NET_WM_STATE:  not found.
```
3.) enable fullscreen for application
4.) disable fullscreen for application
5.) check _NET_WM_STATE of application with `xprop` (and click on application window)
```bash
$ xprop _NET_WM_STATE
_NET_WM_STATE(ATOM) =
```

now the property is set but empty as it should be. Or am I misunderstanding the specs here?

**Comment by lex148** [comment]
Wow, good find. 

I believe you are correct. This should be a minor thing to change however it is  going to take me just a little bit of time to get a fix up. I want to give this a good amount of testing sense it affect so many external applications.

**Comment by lex148** [comment]
Ok, PR is up at [here](https://github.com/leftwm/leftwm/pull/153). I will be testing today. If I don't see anything weird I will merge

**Comment by theguy147** [comment]
closing after being solved with commit a1d287274265ed37992af3a31f4e8c5a13a1eff6
